kartothek/io/dask/delayed.py

```diff
@@ -298,9 +299,17 @@ def read_dataset_as_delayed_metapartitions(
         concat_partitions_on_primary_index=concat_partitions_on_primary_index,
         label_filter=label_filter,
         predicates=predicates,
+        dispatch_by=dispatch_by,
     )
 
-    if concat_partitions_on_primary_index:
+    if concat_partitions_on_primary_index and dispatch_by:
```

Comment:
You should move this check to the `dispatch` function since this is applicable for all pipelines